src/v3/src/components/Form/Form.tsx

```diff
@@ -79,13 +80,23 @@ const Form: FunctionComponent<{
       }
     }
 
-    // submit request
-    onSubmitHandler({
-      includeData: true,
-      includeImmutableData,
-      params,
-      step,
-    });
+    // @ts-expect-error captcha missing from context type
+    const { context: { captcha: captchaConfig } } = currTransaction!;
+    const hasCaptcha = typeof captchaConfig !== 'undefined';
+    if (hasCaptcha) {
+      if (captchaRef?.current) {
+        // launch the captcha challenge
+        captchaRef.current.execute();
+      }
```

Comment:
What happens if `hasCaptcha` is truthy but `captchaRef?.current` is falsy?

Reply:
The form wont submit in that case.  Thats why I made a seperate nested if statement to prevent bypassing the captcha challenge if for some reason the `captchaRef?.current` is falsy.  Or else it would just submit the form anyway.